### PR TITLE
Generate enpoints as types instead of interfaces

### PR DESCRIPTION
**Pull request by ganymedes01**
Generate types as:
```ts
export type EndpointsWithContext<RequestContext> = {
    readonly "/item"?: {
        readonly "post": oar.server.Endpoint<types.Headers$Item$Post, types.Parameters$Item$Post, types.Query$Item$Post, types.RequestBody$Item$Post, types.ShapeOfResponse$Item$Post, RequestContext>;
    };
    readonly "/item/{id}"?: {
        readonly "get": oar.server.Endpoint<types.Headers$Item$Id$Get, types.Parameters$Item$Id$Get, types.Query$Item$Id$Get, types.RequestBody$Item$Id$Get, types.ShapeOfResponse$Item$Id$Get, RequestContext>;
        readonly "head": oar.server.Endpoint<types.Headers$Item$Id$Head, types.Parameters$Item$Id$Head, types.Query$Item$Id$Head, types.RequestBody$Item$Id$Head, types.ShapeOfResponse$Item$Id$Head, RequestContext>;
        readonly "delete": oar.server.Endpoint<types.Headers$Item$Id$Delete, types.Parameters$Item$Id$Delete, types.Query$Item$Id$Delete, types.RequestBody$Item$Id$Delete, types.ShapeOfResponse$Item$Id$Delete, RequestContext>;
    };
};
```

instead of:
```ts
export interface EndpointsWithContext<RequestContext> {
    readonly "/item"?: {
        readonly "post": oar.server.Endpoint<types.Headers$Item$Post, types.Parameters$Item$Post, types.Query$Item$Post, types.RequestBody$Item$Post, types.ShapeOfResponse$Item$Post, RequestContext>;
    };
    readonly "/item/{id}"?: {
        readonly "get": oar.server.Endpoint<types.Headers$Item$Id$Get, types.Parameters$Item$Id$Get, types.Query$Item$Id$Get, types.RequestBody$Item$Id$Get, types.ShapeOfResponse$Item$Id$Get, RequestContext>;
        readonly "head": oar.server.Endpoint<types.Headers$Item$Id$Head, types.Parameters$Item$Id$Head, types.Query$Item$Id$Head, types.RequestBody$Item$Id$Head, types.ShapeOfResponse$Item$Id$Head, RequestContext>;
        readonly "delete": oar.server.Endpoint<types.Headers$Item$Id$Delete, types.Parameters$Item$Id$Delete, types.Query$Item$Id$Delete, types.RequestBody$Item$Id$Delete, types.ShapeOfResponse$Item$Id$Delete, RequestContext>;
    };
};
```

A type such as `Record<string, any>` is compatible with the `type` above, but isn't compatible with the `interface`. This makes it tricky if one is trying to create a generic function that accepts any endpoint declaration object from many different openapi specs

### CODE REVIEW

**Comment by sakari** [comment]
eh. its not assignable? Like [this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChkHJwBcOyA5qSAK4C2ARtMgL66u4zUgJjDohEAFFFIAlCAnRQAJgB4AzmCihyAGiIgAngD4AlDnaSQi5AAdSGLAF4yJMpWQBGZi1y44g07txA) does not work?

**Comment by ganymedes01** [comment]
@sakari [this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChkHJQRwAm6IANgJ7JwD8AXDvoW8WRTcgObMgBXALYAjaKwIBfXNNxhqABxQAhOFGQBeZACUICdFFIAeAM5gooHgBodeg8bMWQ15INHQAfB9y4YAkAhgwBR0ABRQzKpQAJQ4svogZsgKzBhYWth0zJl8yACMksiycKEK0bhAA) doesn't

**Comment by ganymedes01** [comment]
but if you turn `Foo` into a `type` then it works 🤷

**Comment by sakari** [comment]
ah. https://stackoverflow.com/questions/64970414/typescript-assigning-an-interface-or-a-type-to-a-recordstring-string

tldr; because interfaces can be later extended ts gets 🐔 and don't want. Types can't be extended so its ok

Does the type inference when implementing the endpoints work still -- so we do not need to provide types nor do we get any types or something like that in the routes.

**Comment by ganymedes01** [comment]
> Does the type inference when implementing the endpoints work still 
Yeah seems to work the same way whether it's in interface or a type

**Comment by sakari** [comment]
ok then.

**Review by sakari — APPROVED** [review]

**Comment by ganymedes01** [comment]
> tldr; because interfaces can be later extended ts gets 🐔 and don't want. Types can't be extended so its ok

TIL
